pkg/wallet/wallet.go

```diff
@@ -80,6 +80,10 @@ func (w *Wallet) Unprotect(password string, keyPairIndex uint8) error {
 	if err != nil {
 		return fmt.Errorf("intializing block ciphering: %w", err)
 	}
+	// length control on password
+	if len(password) == 0 {
+		return fmt.Errorf("password is empty. try again %w", errors.New("password is empty. try again"))
+	}
```

Comment:
length control on password is done 2 times, we should not do it here but rather before calling Unprotect